site/posts/intake-esm-2020-3-16.ipynb

```diff
@@ -0,0 +1,483 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Search for assets that fulfill our query without `require_all_on`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>experiment_id</th>\n",
+       "      <th>variable_id</th>\n",
+       "      <th>table_id</th>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>source_id</th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>ACCESS-ESM1-5</th>\n",
+       "      <td>3</td>\n",
+       "      <td>2</td>\n",
+       "      <td>1</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>AWI-CM-1-1-MR</th>\n",
+       "      <td>2</td>\n",
+       "      <td>1</td>\n",
+       "      <td>1</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>FGOALS-f3-L</th>\n",
+       "      <td>1</td>\n",
+       "      <td>1</td>\n",
+       "      <td>1</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "               experiment_id  variable_id  table_id\n",
+       "source_id                                          \n",
+       "ACCESS-ESM1-5              3            2         1\n",
+       "AWI-CM-1-1-MR              2            1         1\n",
+       "FGOALS-f3-L                1            1         1"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "col_subset = col.search(**query)\n",
+    "\n",
+    "col_subset.df.groupby(\"source_id\")[\n",
+    "    [\"experiment_id\", \"variable_id\", \"table_id\"]\n",
+    "].nunique()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you can see, the search results above include `source_ids` for which we only have 1 of the two variables, and 1 or 2 or 3 experiments.\n",
+    "\n",
+    "We can tell `intake-esm` to discard any `source_id` that doesn't have both variables `[\"thetao\", \"o2\"]`, all three experiments `[\"historical\", \"ssp245\", \"ssp585\"]` by passing `require_all_on=\"source_id` to the search method:\n"
```

Comment:
I would really emphasize the words "both" and the implied "and" (i.e., the comma):

```suggestion
    "We can tell `intake-esm` to discard any `source_id` that doesn't have *both* variables `[\"thetao\", \"o2\"]` *and* all three experiments `[\"historical\", \"ssp245\", \"ssp585\"]` by passing `require_all_on=\"source_id\"` to the search method:\n"
```